dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -963,7 +970,18 @@ private static void checkHasSourceFiles(WorkflowVersion workflowVersion) {
      */
     public static boolean canAutomaticallyCreateDockstoreOwnedDoi(Workflow workflow, WorkflowVersion workflowVersion) {
         final boolean validPublishedTag = workflow.getIsPublished() && workflowVersion.isValid() && workflowVersion.getReferenceType() == ReferenceType.TAG;
-        return validPublishedTag && !hasExistingDOIForWorkflowVersion(workflowVersion, DoiInitiator.DOCKSTORE);
+        return validPublishedTag && isAutoCreateDockstoreOwnedDoiAllowed(workflow) && !hasExistingDOIForWorkflowVersion(workflowVersion, DoiInitiator.DOCKSTORE);
+    }
+
+    /**
+     * Indicates whether auto-creating a Dockstore owned DOI for a workflow is allowed, based on the configuration file
+     * property <code>gitHubOrgsWithDoiGeneration</code> value. If the allow set is empty, then creating a DOI is allowed for all workflows,
+     * otherwise it's allowed for specified GitHub organizations.
+     * @param workflow
+     * @return
+     */
+    public static boolean isAutoCreateDockstoreOwnedDoiAllowed(Workflow workflow) {
+        return gitHubOrgsWithDoiGeneration.isEmpty() || gitHubOrgsWithDoiGeneration.contains(workflow.getOrganization());
```

Comment:
Still makes me anxious that we could forget to set `gitHubOrgsWithDoiGeneration` and end up generating DOIs for everyone.

Reply:
> Still makes me anxious that we could forget to set `gitHubOrgsWithDoiGeneration` and end up generating DOIs for everyone.

Fair, and I sort of dislike the meaning of the empty list being all orgs; it seems non-intuitive. 2 orgs in the list means 2 orgs are allowed, 1 org in the list means 1 org is allowed, 0 orgs in the list means ~~0~~ all orgs are allowed. :)

We could add an additional property, e.g., enableDoiGeneration, then have this empty list be no orgs.